tests/test_adjoint_jacobian.py

```diff
@@ -605,6 +819,46 @@ def f(params1, params2):
         assert np.allclose(grad_adjoint, grad_fd, atol=tol)
 
 
+@pytest.mark.parametrize(
+    "r_dtype,c_dtype", [[np.float32, np.complex64], [np.float64, np.complex128]]
+)
+def test_qchem_expvalcost_correct(r_dtype, c_dtype):
```

Comment:
This is the test case @AmintorDusko found, which is fixed in this PR.

Reply:
Thank you for the quick fix and test inclusion!